src/herder/TxSetFrame.cpp

```diff
@@ -33,23 +34,42 @@ namespace stellar
 
 using namespace std;
 
+void
+TxSetFrame::computeContentsHash()
```

Comment:
I think if you make this function and sortInHashOrder static functions that return values rather than member functions, you can call them in the class initializer list for `mHash` and `mTxsInHashOrder`, and thereby make those fields `const`. If possible, that would be the strongest version of making this class immutable.

Reply:
This is a great suggestion! 
Fixed.